browser/playlist/android/playlist_page_handler_factory.cc

```diff
@@ -0,0 +1,49 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/playlist/android/playlist_page_handler_factory.h"
+
+#include <utility>
+
+#include "brave/browser/brave_browser_process.h"
+#include "brave/browser/playlist/playlist_service_factory.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/incognito_helpers.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+
+namespace playlist {
+
+// static
+PlaylistPageHandlerFactory* PlaylistPageHandlerFactory::GetInstance() {
+  return base::Singleton<PlaylistPageHandlerFactory>::get();
+}
+
+// static
+mojo::PendingRemote<mojom::PageHandler>
+PlaylistPageHandlerFactory::GetForContext(content::BrowserContext* context) {
+  return static_cast<PlaylistAndroidPageHandler*>(
+             GetInstance()->GetServiceForBrowserContext(context, true))
+      ->MakeRemote();
+}
+
+PlaylistPageHandlerFactory::PlaylistPageHandlerFactory()
+    : BrowserContextKeyedServiceFactory(
+          "PlaylistAndroidPageHandler",
+          BrowserContextDependencyManager::GetInstance()) {
+  DependsOn(playlist::PlaylistServiceFactory::GetInstance());
+}
+
+PlaylistPageHandlerFactory::~PlaylistPageHandlerFactory() = default;
+
+KeyedService* PlaylistPageHandlerFactory::BuildServiceInstanceFor(
+    content::BrowserContext* context) const {
+  auto* profile = Profile::FromBrowserContext(context);
+  playlist::PlaylistServiceFactory::GetInstance()->GetForBrowserContext(
+      context);
+  return new PlaylistAndroidPageHandler(profile);
```

Comment:
this doesn't seem right to me, why is this a keyed service? Also it is not named correctly for a keyed service

Reply:
@bridiver do you have the suggested solution here ? i wanted to keep the implementation  similar to what we have for all other android mojom implementation.

Reply:
if we are doing this anywhere else, those need to change as well. Wallet accesses the actual service as mojo interface, not the page handler.